main.go

```diff
@@ -14,21 +13,30 @@ var log = &logrus.Logger{
 	Out:       os.Stderr,
 	Formatter: new(logrus.TextFormatter),
 	Hooks:     make(logrus.LevelHooks),
-	Level:     logrus.ErrorLevel,
+	Level:     logrus.InfoLevel, // Specifies we only want logs in stdout @ Info or higher
 }
 
 var noCtx = context.Background()
 
-// replyPongToPing is a handler that replies pong to ping messages
-func replyPongToPing(s disgord.Session, data *disgord.MessageCreate) {
-	msg := data.Message
-	if msg.Content != "ping" {
-		return
+// checkErr logs errors if not nil, along with a user-specified trace
+func checkErr(err error, trace string) {
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"trace": trace,
+		}).Error(err)
```

Comment:
nice, did you see if there was a WithField alternative? since there is only one field

Reply:
I think leaving it as WithField is appropriate given it's a template, optimising this code at the cost of functionality and demonstration seems redundant.

Reply:
it's not really an optimization at cost of funcitonality, it's just a different method provided by logrus. The functionality still exists.

Reply:
bad wording, i meant a demonstration of functionality